src/wallet/node_rpc_proxy.cpp

```diff
@@ -217,4 +221,90 @@ boost::optional<std::string> NodeRPCProxy::get_fee_quantization_mask(uint64_t &f
   return boost::optional<std::string>();
 }
 
+const std::vector<cryptonote::COMMAND_RPC_GET_SERVICE_NODES::response::entry> *NodeRPCProxy::get_service_nodes(std::vector<std::string> const &pubkeys, boost::optional<std::string> &failed) const
+{
+  const std::vector<cryptonote::COMMAND_RPC_GET_SERVICE_NODES::response::entry> *result =  &m_service_nodes;
+
+  uint64_t height;
+  failed = get_height(height);
+  if (failed)
+    return result;
+
+  if (m_service_nodes_cached_height != height)
```

Comment:
This caches for a particular height, but what if you specify a different set of nodes?

Reply:
Good point, we should revisit this in the future.